nf_deployments/v1alpha1/nf_deployment_types.go

```diff
@@ -0,0 +1,201 @@
+}
+
+// InterfaceConfig defines the configuration of the interface
+type InterfaceConfig struct {
+	// Name defines the name of the interface
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name" yaml:"name"`
+	// IPv4Config defines the ipv4 configuration of the interface
+	// +optional
+	IPv4Config *IPv4Config `json:"ipv4,omitempty" yaml:"ipv4,omitempty"`
+	// IPv6Config defines the ipv6 configuration of the interface
+	// +optional
+	IPv6Config *IPv6Config `json:"ipv6,omitempty" yaml:"ipv6,omitempty"`
+	// VlanID defines the specific vlan id associated on this interface
+	// +optional
+	VlanID uint16 `json:"vlanID,omitempty" yaml:"vlanID,omitempty"`
+}
+
+// IPv4Config defines the configuration parameters of an ipv4 interface or peer
+type IPv4Config struct {
+	// Address defines the IPv4 address and prefix length in CIDR notation
+	// [IP prefix, range IPv4 with host bits]
+	Address string `json:"address" yaml:"address"`
+	// Gateway defines the IPv4 address associated to the interface as a gateway
+	// +optional
+	Gateway string `json:"gateway,omitempty" yaml:"gateway,omitempty"`
+}
+
+// IPv6Config defines the configuration parameters of an ipv6 interface or peer
+type IPv6Config struct {
+	// Address defines the IPv6 address and prefix length in CIDR notation
+	// [IP prefix, range IPv6 with host bits]
+	Address string `json:"address" yaml:"address"`
+	// Gateway defines the IPv6 address associated to the interface as a gateway
+	// +optional
+	Gateway string `json:"gateway,omitempty" yaml:"gateway,omitempty"`
+}
+
+// A networkInstance is a Layer 3 forwarding construct
+// such as a virtual routing and forwarding (VRF) instance,
+type NetworkInstance struct {
+	// Name defines the name of the network instance
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name" yaml:"name"`
+	// interfaces defines the interfaces associated with the network instance
+	// +optional
+	Interfaces []string `json:"interfaces,omitempty" yaml:"interfaces,omitempty"`
+	// Peers defines the peer configuration associated with the network instance
+	// +optional
+	Peers []*PeerConfig `json:"peers,omitempty" yaml:"peers,omitempty"`
+	// dnns defines the dnns assocated with the network instance
+	// +optional
+	Dnns []*Dnn `json:"dnns,omitempty" yaml:"dnns,omitempty"`
+	// bgp defines the BGP configuration associated with the network instance
+	// +optional
+	Bgp *BGPConfig `json:"bgp,omitempty" yaml:"bgp,omitempty"`
+}
+
+// A Dnn defines the Data Network name defined by 3GPP
+type PeerConfig struct {
+	// Name defines the name of the data network
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// IPv4Config defines the ipv4 configuration of the peer
+	// +optional
+	IPv4Config *IPv4Config `json:"ipv4,omitempty" yaml:"ipv4,omitempty"`
```

Comment:
I think the Go and JSON/YAML names should match

Reply:
fixed